external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -242,9 +242,7 @@ public void nextTuple() {
                     }
                 }
 
-                if (waitingToEmit()) {
-                    emit();
-                }
+                emitIfWaitingNotEmitted();
```

Comment:
If you're removing the waitingToEmit check, I think you should check for null in emitIfWaitingNotEmitted.

Reply:
+1 Nice finding.

Reply:
Done